src/slurm_plugin/fleet_manager.py

```diff
@@ -317,8 +327,15 @@ def _evaluate_launch_params(self, count):
         try:
             common_launch_options = {
                 "SingleInstanceType": self._uses_single_instance_type(),
-                "SingleAvailabilityZone": self._uses_single_az(),  # If using Multi-AZ (by specifying multiple subnets),
-                # set SingleAvailabilityZone to False
+                "SingleAvailabilityZone": (
+                    self._compute_resource_config["Networking"]["SingleAvailabilityZone"]
+                    if self._compute_resource_config["Networking"]["SingleAvailabilityZone"] is not None
+                    and (
+                        self._compute_resource_config.get("AllocationStrategy") == "prioritized"
```

Comment:
Nitpick: Can we check these conditions using a function as we keep checking this above an here